tests/core/test_tracklist.py

```diff
@@ -295,3 +295,41 @@ def test_load_invalid_type(self):
 
     def test_load_none(self):
         self.core.tracklist._load_state(None, None)
+
+
+class TracklistAddTest(unittest.TestCase):
```

Comment:
Add a test for `self.core.tracklist.add(at_position=-2, ...)` to show what a new client using an old version of Mopidy will get?

Reply:
It turns out negative values work just fine, the same way negative indicies work for any regular python list :facepalm: 

Was this behaviour ever officially supported? I'd be surprised if anyone was actually relying on being able to add entries right before the last item in the tracklist, but you never know.

Reply:
If it's not in the documentation then we didn't support it. And nobody should be relying on an undocumented Python-specific implementation detail. So I think it is OK to have this essentially breaking change. We should hold this for until a feature release, maybe even a major release... Although @adamcik's 2nd option would avoid this.

Reply:
I'd definitely prefer to avoid waiting for Mopidy v4 for this change to go through. Instead of adding another keyword argument, I'd prefer to add special string values instead.

Alternatively, it might be better to add a brand new method for this purpose if we think it will be easier to handle proper "play next" when random mode is turned on.